core/state/statedb.go

```diff
@@ -391,42 +403,69 @@ func (s *StateDB) GetCodeSize(addr common.Address) int {
 }
 
 func (s *StateDB) GetCodeHash(addr common.Address) common.Hash {
+	start := time.Now()
+	defer s.markMetrics(start, false)
 	stateObject := s.getStateObject(addr)
 	if stateObject == nil {
 		return common.Hash{}
 	}
 	return common.BytesToHash(stateObject.CodeHash())
 }
 
+func (s *StateDB) markMetrics(start time.Time, reachStorage bool) {
+	goid := cachemetrics.Goid()
+	isSyncMainProcess := cachemetrics.IsSyncMainRoutineID(goid)
+	isMinerMainProcess := cachemetrics.IsMinerMainRoutineID(goid)
+	// record metrics of syncing main process
+	if isSyncMainProcess {
+		totalSyncIOCounter.Inc(time.Since(start).Nanoseconds())
+		l1AccountMeter.Mark(1)
+		if reachStorage {
+			l1StorageMeter.Mark(1)
+		}
+	}
+	// record metrics of mining main process
+	if isMinerMainProcess {
+		totalMinerIOCounter.Inc(time.Since(start).Nanoseconds())
+		minerL1AccountMeter.Mark(1)
+		if reachStorage {
+			minerL1StorageMeter.Mark(1)
+		}
+	}
+
+}
+
 // GetState retrieves a value from the given account's storage trie.
 func (s *StateDB) GetState(addr common.Address, hash common.Hash) common.Hash {
 	start := time.Now()
 	goid := cachemetrics.Goid()
-
 	isSyncMainProcess := cachemetrics.IsSyncMainRoutineID(goid)
 	isMinerMainProcess := cachemetrics.IsMinerMainRoutineID(goid)
 	defer func() {
 		// record metrics of syncing main process
 		if isSyncMainProcess {
 			syncGetDelay := time.Since(start)
-			totalSyncIOCost.Update(syncGetDelay)
-			totalSyncIOCounter.Inc(syncGetDelay.Nanoseconds())
+			totalSyncIOCounter.Inc(time.Since(start).Nanoseconds())
+			getStatetSyncIOCost.Update(syncGetDelay)
+			getStatetSyncIOCounter.Inc(syncGetDelay.Nanoseconds())
 			l1AccountMeter.Mark(1)
 		}
 		// record metrics of mining main process
 		if isMinerMainProcess {
 			minerIOCost := time.Since(start)
-			totalMinerIOCost.Update(minerIOCost)
-			totalMinerIOCounter.Inc(minerIOCost.Nanoseconds())
+			totalMinerIOCounter.Inc(time.Since(start).Nanoseconds())
+			getStatetMinerIOCost.Update(minerIOCost)
+			getStatetMinerIOCounter.Inc(minerIOCost.Nanoseconds())
 			minerL1AccountMeter.Mark(1)
 		}
 	}()
+
 	stateObject := s.getStateObject(addr)
 	if stateObject != nil {
 		if isSyncMainProcess {
 			l1StorageMeter.Mark(1)
 		}
-		if isMinerMainProcess {
+		if isSyncMainProcess {
```

Comment:
Why change this?

Reply:
This variable does not need to be changed, it has been changed back